### PR TITLE
Fix and improve radio buttons on confirmation

### PR DESCRIPTION
**Pull request by 1pretz1**
Currently when a user clicks 'Publish' on the confirmation screen without selecting whether the document has been approved or not we default the document to 'published without review', this is a bad user experience. 

We should ensure that a user confirms a review status when publishing / scheduling a document.

### CODE REVIEW

File: app/controllers/schedule_controller.rb [hunk 1]
```diff
@@ -43,6 +43,12 @@ def confirmation
 
   def schedule
     Edition.find_and_lock_current(document: params[:document]) do |edition|
+      if params[:review_status].nil?
```
Comment (benthorner) [inline app/controllers/schedule_controller.rb]: Could we synchronise the naming (`status` vs. `state`) here?

File: config/locales/en/schedule/confirmation.yml [hunk 2]
```diff
@@ -6,3 +6,6 @@ en:
       review_status:
         reviewed: This content has been reviewed and approved for publication.
         not_reviewed: This content needs to be published urgently but should be reviewed as soon as possible.
+      radio_not_selected:
+        title: No radio button has been selected
```
Comment (benthorner) [inline config/locales/en/schedule/confirmation.yml]: I think the card had some different copy to use - could you double check?

File: spec/features/workflow/publish_without_selection_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+RSpec.feature "Attempting to publish without confirming a review status" do
+  scenario do
+    given_there_is_an_edition_in_draft
+    when_i_visit_the_summary_page
+    and_try_and_publish_without_selecting_a_review_status
```
Comment (benthorner) [inline spec/features/workflow/publish_without_selection_spec.rb]: Also good to synchronise the terminology here (`status` vs. `state`).

File: spec/features/workflow/publish_without_selection_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @edition = create(:edition, :publishable)
+  end
+
+  def when_i_visit_the_summary_page
+    visit document_path(@edition.document)
+  end
+
+  def and_try_and_publish_without_selecting_a_review_status
+    click_on "Publish"
+    stub_publishing_api_publish(@edition.content_id, update_type: nil, locale: @edition.locale)
```
Comment (benthorner) [inline spec/features/workflow/publish_without_selection_spec.rb]: I don't think we need to have this stub here, as hopefully we won't be doing anything :-).

File: spec/features/workflow/schedule_without_selection_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @edition = create(:edition, scheduled_publishing_datetime: Time.current.tomorrow)
+  end
+
+  def when_i_visit_the_summary_page
+    visit document_path(@edition.document)
+  end
+
+  def and_try_and_schedule_without_selecting_a_review_status
+    click_on "Schedule"
+    stub_default_publishing_api_put_intent
```
Comment (benthorner) [inline spec/features/workflow/schedule_without_selection_spec.rb]: Hopefully we shouldn't need a stub here either :-).

File: spec/features/workflow/schedule_without_selection_spec.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+RSpec.feature "Attempting to schedule without confirming a review status" do
+  scenario do
+    given_there_is_an_edition_in_draft
+    when_i_visit_the_summary_page
+    and_try_and_schedule_without_selecting_a_review_status
+    then_an_error_is_displayed
+  end
+
+  def given_there_is_an_edition_in_draft
```
Comment (benthorner) [inline spec/features/workflow/schedule_without_selection_spec.rb]: To be more accurate, could we go with something like `ready_to_schedule`?

File: spec/features/workflow/publish_without_selection_spec.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+RSpec.feature "Attempting to publish without confirming a review status" do
```
Comment (emmabeynon) [inline spec/features/workflow/publish_without_selection_spec.rb]: I would remove "Attempting to" here.  I'd also rename the file to be a bit clearer about what we're testing.  Something like `publish_without_selecting_review_status_spec.rb` perhaps?

File: spec/features/workflow/schedule_without_selection_spec.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+RSpec.feature "Attempting to schedule without confirming a review status" do
```
Comment (emmabeynon) [inline spec/features/workflow/schedule_without_selection_spec.rb]: Similar comment, I would remove "Attempting to" and rename the file to be clearer.

File: spec/features/workflow/publish_without_selection_spec.rb (hunk 7 above)
Comment (benthorner) [reply spec/features/workflow/publish_without_selection_spec.rb]: @emmabeynon we had a long think about this yesterday, and were trying to find a naming pattern that could also be used for the `choose_document_spec`, which does exactly the same kind of check. Taking some inspiration from @1pretz1's choice of locale, how about `publish_status_not_selected_spec.rb`, which could work as `create_document_format_not_selected_spec.rb`?

Comment (emmabeynon) [reply spec/features/workflow/publish_without_selection_spec.rb]: Sounds good, perhapa `publish_review_status_not_selected_spec.rb` just to be really clear?  `create_document_format_not_selected_spec.rb` works for me 👍

Comment (benthorner) [reply spec/features/workflow/publish_without_selection_spec.rb]: Fine with me :-).

Comment (1pretz1) [reply spec/features/workflow/publish_without_selection_spec.rb]: Yep i think Emma's last suggestion is a good file name to use, i'll make the changes! And i'll also open a separate pr for renaming the `choose_document_spec` to follow suite

File: spec/features/workflow/publish_without_selection_spec.rb (hunk 3 above)
Comment (1pretz1) [reply spec/features/workflow/publish_without_selection_spec.rb]: Yep nice spot!

File: spec/features/workflow/publish_without_selection_spec.rb (hunk 4 above)
Comment (1pretz1) [reply spec/features/workflow/publish_without_selection_spec.rb]: Yep you're correct!

File: app/controllers/schedule_controller.rb (hunk 1 above)
Comment (1pretz1) [reply app/controllers/schedule_controller.rb]: Yep, i've gone with `review_status` for both scheduling and publishing controllers

File: config/locales/en/schedule/confirmation.yml (hunk 2 above)
Comment (1pretz1) [reply config/locales/en/schedule/confirmation.yml]: Yep didn't realise content had already been agreed upon 👍

File: spec/features/workflow/schedule_without_selection_spec.rb (hunk 8 above)
Comment (1pretz1) [reply spec/features/workflow/schedule_without_selection_spec.rb]: Yep sure

File: spec/features/workflow/schedule_without_selection_spec.rb (hunk 6 above)
Comment (1pretz1) [reply spec/features/workflow/schedule_without_selection_spec.rb]: Yep ⭐️

File: spec/features/workflow/schedule_without_selection_spec.rb (hunk 5 above)
Comment (1pretz1) [reply spec/features/workflow/schedule_without_selection_spec.rb]: True true true

**Comment by benthorner** [comment]
I've noticed the styling of the error message is different to e.g. requirements issues for images. We should try and make it so errors appear in the same way - could you look into it?

**Comment by 1pretz1** [comment]
@benthorner Yep thats because i was using the error alert component, where as image requirements uses the error summary component. 

Even though the error alert component is used in the application i've made the change as suggested to keep this alert consistent with the other 'requirement' type flashes, although it does certainly look uglier in the code :(

**Pull request merged**